### PR TITLE
Tweaked to compile

### PR DESCRIPTION
**Pull request by julianpistorius**
I will make comments on the important lines.

Feel free to ask questions about particular things which seem confusing.

Overall I get the feeling that you're changing many things at the same time, and then getting into a tangle. I found the most productive way for me to refactor Elm code is to make small, incremental changes, and then letting the compiler guide me to a right solution. When I made too many changes at the same time I ended up drowning in confusing error messages.

### CODE REVIEW

File: src/Main.elm [hunk 1]
```diff
@@ -28,7 +28,7 @@ green =
 
 main : Program () Model Msg
 main =
-    Browser.sandbox { init = init, update = update, view = view, apps = apps }
+    Browser.element { init = init, update = update, view = view, subscriptions = \_ -> Sub.none }
```
Comment (julianpistorius) [inline src/Main.elm]: To talk to the outside world (i.e. HTTP) you have to switch from using [`Browser.sandbox`](https://github.com/elm/browser/blob/d1dd66740615ac872c7cb7ddd9c3aa83d6d288a6/src/Browser.elm#L45):

> Browser.sandbox: Create a “sandboxed” program that cannot communicate with the outside
world.

... to at least [`Browser.element`](https://github.com/elm/browser/blob/d1dd66740615ac872c7cb7ddd9c3aa83d6d288a6/src/Browser.elm#L87):

> Browser.element Unlike a sandbox, an `element` can talk to the outside world...

File: src/Main.elm [hunk 2]
```diff
@@ -74,93 +74,38 @@ init _ =
 -- this is saying GotText can be a result, error, OR string?
 
 
-type AppMsg
-    = GotText (Result Http.Error String)
-
-
-type CounterMsg
+type Msg
```
Comment (julianpistorius) [inline src/Main.elm]: Until your app gets really large it's best to keep your `Msg` type as 'flat' as possible.

File: src/Main.elm [hunk 3]
```diff
@@ -74,93 +74,38 @@ init _ =
...
-                            )
-
-                        -- Return 0
-                        Reset ->
-                            ( { counter = 0
-                              , status = Success fullText
-                              }
-                            , Cmd.none
-                            )
+                    ( { model | status = Success fullText }, Cmd.none )
```
Comment (julianpistorius) [inline src/Main.elm]: This `{model | status = ... }` syntax is a nice way to return a copy of the model where you only modify the `status` field, and leave the rest of the model the way it is.

File: src/Main.elm [hunk 4]
```diff
@@ -74,93 +74,38 @@ init _ =
...
-                              , status = Failure
-                              }
-                            , Cmd.none
-                            )
-
-
+                    ( { model | status = Failure }, Cmd.none )
 
--- APPS
+        Increment ->
```
Comment (julianpistorius) [inline src/Main.elm]: Like I suggest you keep your `Msg` type flat, keep your `update` function as flat as possible as well. You will thank yourself!

File: src/Main.elm [hunk 5]
```diff
@@ -175,13 +120,13 @@ view model =
                 (Element.text
                     (case model.status of
                         Failure ->
-                            Element.text "I was unable to load the text!"
+                            "I was unable to load the text!"
```
Comment (julianpistorius) [inline src/Main.elm]: This was close! You were passing `Element.text String` to `Element.text`, but it expects `String` only. 

Solution is to get rid of `Element.text` at either the top level (outside the `case`), or get rid of it from the individual case branches. I chose the latter in this case.

File: src/Main.elm (hunk 1 above)
Comment (vsoch) [reply src/Main.elm]: I'm guessing that `subscriptions = \_ -> Sub.none` is an inline declaration of apps (or subscriptions that you changed it to) and I looked up the `\_` and I think that is referring to an anonymous function definition?  I didn't realize that subscriptions were a known [thing](https://elmprogramming.com/subscriptions.html) and not just a variable name, so I'll definitely change back to that. I'm going to try doing:

```elm
subscriptions _ =
    Sub.none
```
and keep the subscriptions = subscriptions and see if that works.

File: src/Main.elm (hunk 3 above)
Comment (vsoch) [reply src/Main.elm]: I remember this from the getting started / tutorial pages! I can see how it is useful in practice now.

File: src/Main.elm (hunk 2 above)
Comment (vsoch) [reply src/Main.elm]: Mind blown! I didn't know I could do that (putting the GotText alongside the other states for the counter).

File: src/Main.elm (hunk 4 above)
Comment (vsoch) [reply src/Main.elm]: Woot! I just did this on my own when updating the GotText result, and looks like I did it right :)

**Comment by vsoch** [comment]
I tried adding you to the other PR via the [Coauthored by](https://help.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors) but it didn't work, so I resolved conflicts here and am going to merge the trivial commit - it's important to me that your help is represented in the commit history. For next time - what is your email associated with your GitHub so coauthored by works?

File: src/Main.elm (hunk 1 above)
Comment (julianpistorius) [reply src/Main.elm]: Yeah, `\_ -> Sub.none` is a short-hand to say there are no subscriptions for this element. 

The `\x -> <whatever>` is Elm's syntax for an anonymous function (aka lambdas in Python): https://elm-lang.org/docs/syntax#functions

The underscore as parameter to a function means you don't care about the actual value of that parameter.

I usually prefer named functions like you did with `subscriptions`. Nice job! Good instinct for readability and extensibility there. 💪 

And yes, subscriptions is a thing. The `Browser.element` function takes one parameter - a record with specific fields, one of which _has_ to be `subscriptions`:

https://github.com/elm/browser/blob/d1dd66740615ac872c7cb7ddd9c3aa83d6d288a6/src/Browser.elm#L104

When you changed the `subscriptions` field to `apps` you changed the type of the record, and the compiler would have complained.

More info about Elm records:

- https://elm-lang.org/docs/syntax#records
- https://elm-lang.org/docs/records

**Comment by julianpistorius** [comment]
Thanks @vsoch!

> For next time - what is your email associated with your GitHub so coauthored by works?

Can you see if this one works: julianpistorius@users.noreply.github.com

**Comment by vsoch** [comment]
You got it! I'm having a lot of fun working on this, I really appreciate your thoughtful responses and time. I'll keep saying that if I haven't already. :)

**Comment by julianpistorius** [comment]
My pleasure @vsoch. Glad you're having fun and learning!

I'm learning a lot through this coaching process as well.

**Pull request merged**